cmd/openshift-tests/provider.go

```diff
@@ -139,6 +139,19 @@ func decodeProvider(provider string, dryRun, discover bool, clusterState *exutil
 		if err := json.Unmarshal([]byte(provider), config); err != nil {
 			return nil, fmt.Errorf("provider must decode into the ClusterConfig object: %v", err)
 		}
+
+		coreClient, err := e2e.LoadClientset(true)
+		if err != nil {
+			return nil, err
+		}
+		isMicroShift, err := exutil.IsMicroShiftCluster(coreClient)
```

Comment:
I would do this in "DiscoverClusterState" and not "decodeProvider".

The provider is irrelevant

Reply:
Agree with Ben, even though the name might be misleading because its more discovery than decoding.